ext_mongodb.php

```diff
@@ -620,6 +620,78 @@ class BulkWriteException extends WriteException {}
 /* {{{ BSON and Serialization Classes */
 namespace MongoDB\BSON;
 
+/* {{{ Interfaces */
+interface TypeWrapper
+{
+	static public function createFromBSONType(\MongoDB\BSON\Type $type) : \MongoDB\BSON\TypeWrapper;
+	public function toBSONType() : \MongoDB\BSON\Type;
+}
+
+interface BinaryInterface
+{
+	public function __construct(string $data, int $type);
```

Comment:
Is there a particular reason to enforce a `__construct()` signature, or `__debugInfo()` for that matter? In any of our code (e.g. PHPLIB), we'll end up using the original BSON classes directly, so this doesn't affect us.

For the purposes of argument, I can imagine a custom Binary wrapper defaulting its `$type` argument to a particular value. Alternatively, folks may have a Regex constructed from a single parameter like MongoRegex (e.g. `/pattern/flags`).